erpnext/schools/doctype/student_group/student_group.js

```diff
@@ -30,19 +48,46 @@ frappe.ui.form.on("Student Group", {
 			});
 		}
 	},
+	
+	group_based_on: function(frm) {
+		if (frm.doc.group_based_on == "Batch") {
+			frm.doc.course = null;
+		}
+		else if (frm.doc.group_based_on == "Course") {
+			frm.doc.program = null;
+			frm.doc.batch = null;
+		}
+	},
 
-	onload: function(frm) {
-		frm.set_query("academic_term", function() {
-			return {
-				"filters": {
-					"academic_year": (frm.doc.academic_year)
+	students: function(frm) {
+		console.log("in students");
+	},
+
+	update_students: function(frm) {
+		if (frm.doc.students.length === 0) {
+			frm.doc.next_group_roll_number = 1;
+		}
+		frappe.call({
+			method: "update_students",
+			doc:frm.doc,
```

Comment:
I think you need to pass only 4 fields. There is no need to check is student already fetched  or not in the server side, it can be done in client side as well while adding.